main.go

```diff
@@ -86,30 +94,128 @@ func isDockerCommand(kw string) bool {
 	return false
 }
 
-func completer(d prompt.Document) []prompt.Suggest {
-	word := d.GetWordBeforeCursor()
+//DockerHubResult : Wrap DockerHub API call
+type DockerHubResult struct {
+	PageCount        *int             `json:"num_pages,omitempty"`
+	ResultCount      *int             `json:"num_results,omitempty"`
+	ItemCountPerPage *int             `json:"page_size,omitempty"`
+	CurrentPage      *int             `json:"page,omitempty"`
+	Query            *string          `json:"query,omitempty"`
+	Items            []DockerHubImage `json:"results,omitempty"`
+}
+
+//DockerHubImage : Wrap image results of DockerHub API call
+type DockerHubImage struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	StarCount   int    `json:"star_count,omitempty"`
+	IsTrusted   *bool  `json:"is_trusted,omitempty"`
+	IsAutomated bool   `json:"is_automated,omitempty"`
+	IsOffical   *bool  `json:"is_official,omitempty"`
+}
+
+func imageFetchCompleter(imageName string, count int) []prompt.Suggest {
+	url := url.URL{
+		Scheme:   "https",
+		Host:     "registry.hub.docker.com",
+		Path:     "/v1/search",
+		RawQuery: "q=" + url.PathEscape(imageName) + "&n=" + strconv.Itoa(count),
+	}
+	if imageName == "" {
+		url.Path = "/v2/repositories/library"
+		url.RawQuery = "page=1&page_size=" + strconv.Itoa(count)
+	}
+	client := &http.Client{}
```

Comment:
We should add timeout here
```suggestion
	client := &http.Client{
	    Timeout: 2 * time.Second,
	}
```

Reply:
Totally agree!

Reply:
By the way, instead of writing whole image search process what about to use docker client?
Like this:
```
ctx, _ := context.WithTimeout(context.Background(), 2*time.Second)

res, err := dockerClient.ImageSearch(ctx, "<word>", types.ImageSearchOptions{Limit: 10})
```

Reply:
I've written below function but the performance didn't make me happy. Did you try it?
 
```golang
func contextImageFetchCompleter(imageName string, count int) []prompt.Suggest {
	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
	defer cancel()
	res, err := dockerClient.ImageSearch(ctx, imageName, types.ImageSearchOptions{Limit: count})
	if err != nil {
		return nil
	}

	suggestions := []prompt.Suggest{}
	for _, s := range res {
		description := "Not Offical"
		if s.IsOfficial {
			description = "Offical"
		}
		suggestions = append(suggestions, prompt.Suggest{Text: s.Name, Description: "(" + description + ") " + s.Description})
	}
	return suggestions
}
```

Reply:
I could not faced with performance issue. Actually it does nearly same thing behind image search function just with several additions. If you claim that it will be more performant friendly to go without docker client, we can merge your changes as is 👍

Reply:
I made small changes. When the pull command doesn't have any keyword, images are pulling from Docker Hub API. Otherwise, the shell will use context. If it's OK, you can merge it 👍